### ISSUE TITLE
Different ways of total epoch loss calculation in train and validation. Is it correct?

### ISSUE THREAD

**Issue by MistyDi** [zylo117/Yet-Another-EfficientDet-Pytorch#288]
Hi, thanks for huge work!
I noticed that total epoch loss-function is calculated different ways, if compare train part and validation part of [`train.py`](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/blob/master/train.py) code. Is it correct?

In train: total epoch loss is **mean of array, that contain sums** of every batches' cls_loss' mean and reg_loss' mean.
![image](https://user-images.githubusercontent.com/47452727/81596910-80ff1180-93cd-11ea-8dad-39672effd4a8.png)

In val: total epoch loss is **sum of means of two arrays**: one is batches' mean of cls_losses, another is batches' means of reg_losses.
![image](https://user-images.githubusercontent.com/47452727/81598044-5d3ccb00-93cf-11ea-9f53-52815b15ed27.png)

**Comment by zylo117** [comment]
The calculations are different because most of them are mainly from <https://github.com/signatrix/efficientdet>. But the result should be the same if given the same input.

**Comment by MistyDi** [comment]
You are right. The result is the same. Thanks for comment!
Also i checked it to prove that the results of these two ways of mean calculation is the same:

```
import numpy as np

train_epoch_loss = []
val_cls_loss = []
val_reg_loss = []

for epoch in range(5):
    print('Epoch: ', epoch)
    for batch_id in range(200):
        cls_loss = np.random.uniform(0, 1, size=(1, 6))
        reg_loss = np.random.uniform(0, 1, size=(1, 6))
        
        cls_loss = np.mean(cls_loss)
        reg_loss = np.mean(reg_loss)
        
        train_epoch_loss.append(cls_loss + reg_loss)
        
        val_cls_loss.append(cls_loss)
        val_reg_loss.append(reg_loss)
        
    print('Total loss train-like:', np.mean(train_epoch_loss))
    print('Total loss val-like:  ', np.mean(val_cls_loss) + np.mean(val_reg_loss), end='\n\n')
```
Output:
```Epoch:  0
Total loss train-like: 0.9897764435229186
Total loss val-like:   0.9897764435229185

Epoch:  1
Total loss train-like: 0.9975985093202712
Total loss val-like:   0.997598509320271

Epoch:  2
Total loss train-like: 0.998532088797012
Total loss val-like:   0.998532088797012

Epoch:  3
Total loss train-like: 1.0009208336885331
Total loss val-like:   1.0009208336885331

Epoch:  4
Total loss train-like: 0.9990762343861193
Total loss val-like:   0.9990762343861194
```
